accession-commons-jpa/src/test/java/uk/ac/ebi/ampt2d/test/configuration/TestJpaDatabaseServiceTestConfiguration.java

```diff
@@ -50,6 +53,9 @@
 @EnableJpaRepositories(basePackages = {"uk.ac.ebi.ampt2d.test.persistence",
         "uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.repositories"
 })
+@AutoConfigureCache
+@AutoConfigureDataJpa
+@AutoConfigureTestEntityManager
```

Comment:
Why not [DataJpaTest](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/orm/jpa/DataJpaTest.html) instead of these 3 annotations?

Reply:
Because DataJpaTest introduces all changes inside a transaction that doesn't play well with the system, as it doesn't expect to be inside another transaction and the behavior breaks completely.